src/braket/circuits/compiler_directive.py

```diff
@@ -41,8 +43,43 @@ def ascii_symbols(self) -> Tuple[str, ...]:
         """Tuple[str, ...]: Returns the ascii symbols for the compiler directive."""
         return self._ascii_symbols
 
-    def to_ir(self, *args, **kwargs):
-        raise NotImplementedError("to_ir has not been implemented yet.")
+    def to_ir(
+        self,
+        target: QubitSet = None,
+        ir_type: IRType = IRType.JAQCD,
+        qubit_reference_format: str = "${}",
+        **kwargs,
+    ):
+        """Returns IR object of the compiler directive.
+
+        Args:
+            target (QubitSet): target qubit(s). Defaults to None
+            ir_type(IRType) : The IRType to use for converting the compiler directive object to its
+                IR representation. Defaults to IRType.JAQCD.
+            qubit_reference_format (str): The string format to use for referencing the qubits
+                within the gate. Defaults to "${}" for referencing qubits physically.
+            **kwargs: Keyword arguments
+
+        Returns:
+            IR object of the compiler directive.
+
+        Raises:
+            ValueError: If the supplied `ir_type` is not supported.
+        """
+        if ir_type == IRType.JAQCD:
+            return self.to_jaqcd()
+        elif ir_type == IRType.OPENQASM:
+            return self.to_openqasm()
+        else:
+            raise ValueError(f"Supplied ir_type {ir_type} is not supported.")
+
+    def to_jaqcd(self) -> Any:
```

Comment:
should these be private?

Reply:
This is something I wasn't sure about. It would be nice to have them private, but then having an interface define private methods that the subclasses **need** to implement felt weird. I can change to private if you think that's fine.

Reply:
Yeah, I can see it either way. Technically, subclasses don't need to implement these methods (they can implement to_ir() instead), but they can choose to rely on the fact that the parent is calling these (what would be protected in C++) private functions. We definitely don't want customers calling these directly, right? That makes me lean toward making them private, but it's up to you.

Reply:
Updated to private